Migrate Gradle plugin v1 to v2 (#3602)
getsentry/sentry-docs: src/platforms/android/migration.mdx

  - /platforms/android/migrate/
---

> ## Migrating from `io.sentry:sentry-android-gradle-plugin 1.x` to `io.sentry:sentry-android-gradle-plugin 2.0.0`

Reviewer: ```suggestion
## Migrating Android Grade Plugin From 1.x to 2.x 
```

> The `io.sentry.android.gradle` >= `2.0.0` publishes the Gradle Plugin Marker to Maven Central as well, so now you don't need to set the `classpath` anymore.

Reviewer: ```suggestion
The `io.sentry.android.gradle` >= `2.0.0` publishes the Gradle Plugin Marker to Maven Central. As a result, you no longer need to set the `classpath`.
```

_Old_:

```groovy
buildscript {
    repositories {
        mavenCentral()
    }

    dependencies {
        classpath 'io.sentry:sentry-android-gradle-plugin:{version}'
    }
}

apply plugin: 'io.sentry.android.gradle'
```

_New_:

```groovy
plugins {
    id "io.sentry.android.gradle" version "{version}"
}

buildscript {
    repositories {
        mavenCentral()
    }
}
```

```kotlin
plugins {
    id("io.sentry.android.gradle") version "{version}"
}

buildscript {
    repositories {
        mavenCentral()
    }
}
```

> The Sentry Android Gradle Plugin has been rewritten in Kotlin and because of that, the Groovy or Kotlin DSL syntax has slightly changed.

Reviewer: ```suggestion
The Sentry Android Gradle Plugin has been rewritten in Kotlin. As a result, the Groovy or Kotlin DSL syntax has slightly changed.
```

The `autoProguardConfig` flag has been removed, the Android SDK >= `2.0.0` version already merges the ProGuard rules automatically.

_Old_:

```groovy
sentry {
    autoProguardConfig false
    autoUpload true
    uploadNativeSymbols false
    includeNativeSources false
}
```

_New_:

```groovy
sentry {
    autoUpload = true
    uploadNativeSymbols = false
    includeNativeSources = false
}
```

```kotlin
sentry {
    autoUpload.set(true)
    uploadNativeSymbols.set(false)
    includeNativeSources.set(false)
}
```

> ## Migrating from `io.sentry:sentry-android 4.3.0` to `io.sentry:sentry-android 5.0.0`

Reviewer: ```suggestion
## Migrating Sentry Android from 4.3 to 5.0 
```

</application>
```

> ## Migrating from `io.sentry:sentry-android 4.1.0` to `io.sentry:sentry-android 4.2.0`

Reviewer: ```suggestion
## Migrating Sentry Android from 4.1.0 to 4.2.0
```

`operation` is now a required property of the `SentryTransaction` and `SentrySpan`. Whenever a transaction or span is started, the value for `operation` must be provided:

```java
Sentry.startTransaction("transaction-name", "operation-name");
```

> ## Migrating from `io.sentry:sentry-android 2.x` to `io.sentry:sentry-android 3.x`

Reviewer: ```suggestion
## Migrating Sentry Android from 2.x to 3.x
```

</application>
```

> ## Migrating from `io.sentry:sentry-android 1.x` to `io.sentry:sentry-android 2.x`

Reviewer: ```suggestion
## Migrating Sentry Android from 1.x to  2.x
```

### Sentry Gradle Plugin

> Disable the `autoProguardConfig` flag from the Sentry Gradle Plugin, `io.sentry:sentry-android` >= `2.0.0` already does it automatically.

Reviewer: ```suggestion
Disable the `autoProguardConfig` flag from the Sentry Gradle Plugin since `io.sentry:sentry-android` >= `2.0.0 does this automatically.
```
